src/main/java/io/rcktapp/api/handler/firehose/DeliveryStreamNameSpliterator.java

```diff
@@ -0,0 +1,55 @@
+package io.rcktapp.api.handler.firehose;
+
+import com.amazonaws.services.kinesisfirehose.AmazonKinesisFirehose;
+import com.amazonaws.services.kinesisfirehose.model.ListDeliveryStreamsRequest;
+import com.amazonaws.services.kinesisfirehose.model.ListDeliveryStreamsResult;
+
+import java.util.List;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class DeliveryStreamNameSpliterator implements Spliterator<String> {
+    private final AmazonKinesisFirehose firehoseClient;
+    private String last = null;
+
+    private DeliveryStreamNameSpliterator(AmazonKinesisFirehose firehoseClient) {
+        this.firehoseClient = firehoseClient;
+    }
+
+    static Stream<String> stream(AmazonKinesisFirehose firehoseClient) {
+        return StreamSupport.stream(new DeliveryStreamNameSpliterator(firehoseClient), false);
+    }
+
+
+    @Override
+    public boolean tryAdvance(Consumer<? super String> action) {
```

Comment:
This is the core of a spliterator implementation - tryAdvance. "When I advance to the next element and don't have any, what do I do?" - it provides a Consumer (of the next elements) and the return value is "is there any more to look for"